### PR TITLE
Use bingo to pin all Go dependencies

### PR DESCRIPTION
**Pull request by metalmatze**
Use [bingo](https://github.com/bwplotka/bingo) to pin all development dependencies and add a few Makefile targets to build all deps and targets themselves.

Works like a charm on CI and builds in 30s. :tada: 

/cc @bwplotka @krasi-georgiev

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/metalmatze/slo-libsonnet
```
Comment (bwplotka) [inline go.mod:1]: Do we ned this? I don't think so

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,34 @@
 all: examples
```
Comment (bwplotka) [inline Makefile:1]: There is much better way! 

Just `go get github.com/bwplotka/bingo@v0.2.1` (or bingo get)

then run `bingo get` and here you can put `include .bingo/Variables.mk`. Then the FUN begins when you can just use variables, no need for manual copy of  all those: 
```
.bingo/jb:
	go build -modfile .bingo/jb.mod -o .bingo/jb github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb

.bingo/jsonnet:
	go build -modfile .bingo/jsonnet.mod -o .bingo/jsonnet github.com/google/go-jsonnet/cmd/jsonnet

.bingo/jsonnetfmt:
	go build -modfile .bingo/jsonnetfmt.mod -o .bingo/jsonnetfmt github.com/google/go-jsonnet/cmd/jsonnetfmt

.bingo/gojsontoyaml:
	go build -modfile .bingo/gojsontoyaml.mod -o .bingo/gojsontoyaml github.com/brancz/gojsontoyaml
```

:hugs: 

Similar to: https://gitlab.cee.redhat.com/observatorium/configuration/merge_requests/87

**Comment by metalmatze** [comment]
> Nice, I think you can improve Makefile a lot though. SImplify with Bingo even more (:

I actually quite like repeating Makefiles, rather than magic I like using multi cursor edits :smile: 
Keeps it obvious to people who just skim at it.

File: go.mod (hunk 1 above)
Comment (metalmatze) [reply go.mod:1]: It failed to initialize without the root go.mod file at least. We might be able to remove it now, but I had to have it for the first bingo deps.

**Comment by bwplotka** [comment]
Up to you, make sense - will work well

**Pull request merged**